### ISSUE TITLE
Why is Get-AzAccessToken going to start returning a SecureString?

### ISSUE THREAD

**Issue by jackmtpt** [Azure/azure-powershell#26847]
### Description

Other Microsoft code is actively migrating _away_ from `SecureString` , e.g. https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2437.

Microsoft's own documentation about SecureString recommends that it is not used: https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-security-securestring

Apparently `Az` version 14 is going to ignore this guidance and make a breaking change to use SecureString anyway:

```
WARNING: Upcoming breaking changes in the cmdlet 'Get-AzAccessToken' :
The Token property of the output type will be changed from String to SecureString. Add the [-AsSecureString] switch to avoid the impact of this upcoming breaking change.
- The change is expected to take effect in Az version : '14.0.0'
- The change is expected to take effect in Az.Accounts version : '4.0.0'
```

Why?

### Script or Debug output

```PowerShell

```

### Environment data

```PowerShell

```

### Module versions

```PowerShell

```

### Error output

```PowerShell

```

**Comment by isra-fel** [comment]
Hi @jackmtpt thanks for reaching out. Long story short, we promote secure string to prevent the risk of unintentional exposure of plain text string, not for cryptographic reasons.

PowerShell scripts are often executed in automation systems. Some of them lack comprehensive access control over logs, which when combined with unsafe programming practices, leads to the exposure of secrets. That's the problem we want to address with SecureString. Even if a SecureString is logged, it's never in plain text, hence free of risk.

The reason why SecureString is unrecommended in .NET is that it's not safe cryptographically. Since Azure PowerShell is not relying on it to encrypt those data, it is unnecessary to move away from SecureString as well.

**Comment by jackmtpt** [comment]
There's better ways to do that though aren't there? e.g. via the [Types.ps1xml](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_types.ps1xml?view=powershell-7.4#creating-new-typesps1xml-files) files that specify which fields are shown by default when an object is written to the pipeline. That would prevent the token getting printed to the terminal/logs accidentally unless someone explicitly accesses `.Token`.

**Comment by isra-fel** [comment]
Yes that's one option. But there are limitations that (a) as you said if someone explicitly `.Token` or does customized format with `| Format-*` it won't work. (b) `format.ps1xml` doesn't apply to the result of jobs `$j = start-job { get-azaccesstoken }; $j | receive-job`.

**Comment by davidbviana** [comment]
Recently, the deprecation went into effect, so our workaround was the following:
For some reason, we had:
`$token = ConvertTo-SecureString (Get-AzAccessToken -Resource $resource_id).Token -AsPlainText -Force`
And that became (by dropping the `ConvertTo-SecureString` call):
`$token = (Get-AzAccessToken -Resource $resource_id).Token`

**Comment by isra-fel** [comment]
Adding to David, depending on your purpose for the token -

1. If whoever consumes the token also accepts a `SecureString`, that's perfect - you can simply remove the redundant code for conversion, as shown in David's example
2. If you need the token to make a web request to Azure, consider [`Invoke-AzRestMethod`](https://learn.microsoft.com/en-us/powershell/module/az.accounts/invoke-azrestmethod?view=azps-14.1.0#example-1) which frees you completely from handling the token by yourself.
3. If you absolutely need the token in plaintext - follow the [migration guide](https://github.com/Azure/azure-powershell/blob/main/documentation/migration-guides/Az.14.0.0-migration-guide.md#get-azaccesstoken) or use [`ConvertFrom-SecureString`](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.security/convertfrom-securestring?view=powershell-7.5#example-4-convert-a-secure-string-directly-to-a-plaintext-string) (PS 7 required).
